From 85a6e0a5b5d36612ec4525f3b07f8b48c5cbb4b5 Mon Sep 17 00:00:00 2001
From: Narendhar Reddy <quic_nare@quicinc.com>
Date: Fri, 12 Jul 2024 09:19:10 +0530
Subject: [PATCH] Revert "bus: mhi: host: Skip MHI reset if device is in RDDM"

This reverts commit cabce92dd805945a090dc6fc73b001bb35ed083a.

Revert this change as we are getting error during mhi
power down/power up with coldboot enabled in ath driver.

Error log:
      [69788.680857] mhi mhi0: reg:ERROR_CODE value:0xef110000
      [69788.683827] mhi mhi0: reg:ERROR_DBG1 value:0x0
      [69788.689020] mhi mhi0: reg:ERROR_DBG2 value:0x0
      [69788.693373] mhi mhi0: reg:ERROR_DBG3 value:0x0

Change-Id: I995798ae4863e48992c9c092ea024f89c4395c7e
Signed-off-by: Narendhar Reddy <quic_nare@quicinc.com>
---
 drivers/bus/mhi/host/pm.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/drivers/bus/mhi/host/pm.c b/drivers/bus/mhi/host/pm.c
index 5001a15e0800..03417fb5e664 100644
--- a/drivers/bus/mhi/host/pm.c
+++ b/drivers/bus/mhi/host/pm.c
@@ -481,10 +481,6 @@ static void mhi_pm_disable_transition(struct mhi_controller *mhi_cntrl)
 
 	/* Trigger MHI RESET so that the device will not access host memory */
 	if (!MHI_PM_IN_FATAL_STATE(mhi_cntrl->pm_state)) {
-		/* Skip MHI RESET if in RDDM state */
-		if (mhi_cntrl->rddm_image && mhi_get_exec_env(mhi_cntrl) == MHI_EE_RDDM)
-			goto skip_mhi_reset;
-
 		dev_dbg(dev, "Triggering MHI Reset in device\n");
 		mhi_set_mhi_state(mhi_cntrl, MHI_STATE_RESET);
 
@@ -510,7 +506,6 @@ static void mhi_pm_disable_transition(struct mhi_controller *mhi_cntrl)
 		}
 	}
 
-skip_mhi_reset:
 	dev_dbg(dev,
 		 "Waiting for all pending event ring processing to complete\n");
 	mhi_event = mhi_cntrl->mhi_event;
-- 
2.34.1

